gson/src/main/java/com/google/gson/internal/ConstructorConstructor.java

```diff
@@ -370,29 +370,29 @@ private static boolean hasStringKeyType(Type mapType) {
     return GsonTypes.getRawType(typeArguments[0]) == String.class;
   }
 
-  private static ObjectConstructor<? extends Map<? extends Object, Object>> newMapConstructor(
+  private static ObjectConstructor<? extends Map<?, Object>> newMapConstructor(
       Type type, Class<?> rawType) {
     // First try Map implementation
     /*
      * Legacy special casing for Map<String, ...> to avoid DoS from colliding String hashCode
      * values for older JDKs; use own LinkedTreeMap<String, Object> instead
      */
     if (rawType.isAssignableFrom(LinkedTreeMap.class) && hasStringKeyType(type)) {
-      return () -> new LinkedTreeMap<>();
+      return LinkedTreeMap::new;
```

Comment:
```suggestion
      // Must use lambda instead of method reference (`LinkedTreeMap::new`) here, otherwise this
      // causes an exception when Gson is used by a custom system class loader, see
      // https://github.com/google/gson/pull/2864#issuecomment-3528623716
      return () -> new LinkedTreeMap<>();
```